etw/sample/sample.go

```diff
@@ -0,0 +1,53 @@
+// Shows a sample usage of the ETW logging package.
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+
+	"github.com/Microsoft/go-winio/etw"
+	"github.com/sirupsen/logrus"
+
+	"golang.org/x/sys/windows"
+)
+
+func callback(sourceID *windows.GUID, isEnabled uint32, level byte, matchAnyKeyword uint64, matchAllKeyword uint64, filterData uintptr) {
+	fmt.Printf("Callback: isEnabled=%d, level=%d, matchAnyKeyword=%d\n", isEnabled, level, matchAnyKeyword)
+}
+
+func main() {
+	providerID := windows.GUID{0xdd2062c6, 0x5d1b, 0x4a0f, [8]uint8{0xbd, 0xb9, 0x22, 0x28, 0xbc, 0xb1, 0x07, 0x7c}}
+
+	provider, err := etw.NewProvider("TestProvider", &providerID, callback)
+	if err != nil {
+		logrus.Error(err)
+		return
+	}
+	defer func() {
+		if err := provider.Close(); err != nil {
+			logrus.Error(err)
+		}
+	}()
+
+	reader := bufio.NewReader(os.Stdin)
+
+	fmt.Println("Press enter to log an event")
+	reader.ReadString('\n')
+
+	event := etw.NewEvent("TestEvent", etw.NewEventDescriptor())
```

Comment:
These could really be test cases. I think it would be nice to show a test case for the various `InTypes` which would make this more clear how to handle it. My guess is that zero consumers will write this code. Instead they will write.

```go
logrus.WithFields(logrus.Fields{
    "id", idString,
    "value", valueInt32,
    "time", time.Now(),
}.Debug("Trace Point")
```
And what we want to see is how the hook writes a "String, int32, time -> int?" etc.

Reply:
I can move this to be test code instead of a sample. I'm not sure there's anything meaningful to check against to see if the test succeeds (other than validating an error didn't get returned), though.

Reply:
I think we need to create a test listener so that we can verify that each event we generate is written as we expect into the platform. Would be really nice to know this code passes all tests. If we need to fake this we can at the syscall layer but I thought we could easily make a listener and just exercise the product in the test.